common/strings.c

```diff
@@ -331,3 +331,43 @@ TDNFTrimSuffix(
 error:
     goto cleanup;
 }
+
+uint32_t
+TDNFStringEndsWith(
+    char* pszSource,
+    const char* pszSuffix
+    )
+{
+    int nSourceStrLen = 0, nSuffixStrLen = 0;
+    uint32_t dwError = 0;
+    int ret = 0;
+
+    if (!pszSource || !pszSuffix)
```

Comment:
should this be IsNullOrEmptyString?

Reply:
Yeah, modified this to IsNullOrEmptyString. Thats should be fine.